squashfs-mount.c

```diff
@@ -120,8 +133,23 @@ int main(int argc, char **argv) {
   if (mnt_context_set_target(cxt, mountpoint) != 0)
     exit_with_error("Failed to set target\n");
 
-  if (mnt_context_mount(cxt) != 0)
+  if (flock(squashfs_fd, LOCK_EX) != 0)
+    exit_with_error("Could not acquire a lock on the squashfs file\n");
+
+  int mount_exit_code = mnt_context_mount(cxt);
+  if (mount_exit_code != 0) {
+    mnt_context_get_excode(cxt, mount_exit_code, err_buf, sizeof(err_buf));
+    const char *tgt = mnt_context_get_target(cxt);
+    if (*err_buf != '\0' && tgt != NULL)
+      exit_with_error("%s: %s\n", tgt, err_buf);
```

Comment:
Should the lock be released before exiting, so that other processes don't dead lock?

Reply:
I believe it's automatically unlocked when the file descriptor is closed, which should happen when the process terminates

Reply:
From the man page for flock:
```
       Furthermore, the lock is released either by an explicit LOCK_UN  operation  on  any  of  these  duplicate  file
       descriptors, or when all such file descriptors have been closed.
```
Which backs you up. But it is still cleaner to close the lock explicitly.